### PR TITLE
Log more in flaky transaction_result_does_not_affect_bankhash test

### PR DESCRIPTION
**Pull request by steviez**
#### Problem
The test is flaky, but in the current state, it is impossible to determine what might be failing as we only report equality on the bank hashes.

#### Summary of Changes
So, get and compare all pieces that go into bank hash computation

### CODE REVIEW

**Review by brooksprumo — COMMENTED** [review]
I'm not really familiar with part of the code, so I'm not confident what should be right/wrong here. But I agree with the PR description in general. How much of a review are you looking for?

**Comment by steviez** [comment]
> I'm not really familiar with part of the code, so I'm not confident what should be right/wrong here. But I agree with the PR description in general. How much of a review are you looking for?

Not much ha. This is kind of an in-between review; I initially added you because I was making changes to AccountsDb before I realized I could simplify. And Pankaj has his name on git diff for this test.

My update takes advantage of the same code that we use to generate bank hash details file when a node realizes it has diverged. So, I get a dump of that (accounts delta, last_blockhash, written accounts, etc) in the assert statement. After typing this out, I realize that Ashwin is pretty familiar with these "bank hash debug files" so he might get a good candidate for review if you don't want to just rubber-stamp and/or spend time digging just for this minimal PR

For example, if I change `assert_eq` to `assert_ne` to make it fail, I get this
```
assertion `left != right` failed: bank hash did not mismatch for error: InvalidArgument
  left: SlotDetails { slot: 0, bank_hash: "J1WR5g3uGj3MAMt2r18cLxTjgND59Bv8z7AfQpWJrpus", bank_hash_components: Some(BankHashComponents { parent_bank_hash: "11111111111111111111111111111111", accounts_delta_hash: "2PxWbMDujdEL4Rs52i8BVZc23MpQaFXvpga5VGrxiCkc", ...
  right: SlotDetails { slot: 0, bank_hash: "J1WR5g3uGj3MAMt2r18cLxTjgND59Bv8z7AfQpWJrpus", bank_hash_components: Some(BankHashComponents { parent_bank_hash: "11111111111111111111111111111111", accounts_delta_hash: "2PxWbMDujdEL4Rs52i8BVZc23MpQaFXvpga5VGrxiCkc", ...
```

**Comment by brooksprumo** [comment]
Sounds good. I'll defer to the others, thanks!

**Review by pgarg66 — DISMISSED** [review]
Looks good to me. Let's wait for Ashwin's review as well.

File: ledger/src/blockstore_processor.rs
Comment (AshwinSekar) [inline ledger/src/blockstore_processor.rs]: nit: should be "bank hash mismatched for error:"

**Review by AshwinSekar — DISMISSED** [review]
LGTM

File: ledger/src/blockstore_processor.rs
Comment (steviez) [reply ledger/src/blockstore_processor.rs]: Nice catch, fixed it in https://github.com/anza-xyz/agave/pull/3549/commits/f45b33d6632e3d83546835d2ef40aef342d5b259

**Review by AshwinSekar — APPROVED** [review]

**Pull request merged**
